pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -155,6 +155,9 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) && !localFeatureGate.Enabled(features.GracefulNodeShutdown) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: Specifying ShutdownGracePeriod or ShutdownGracePeriodCriticalPods requires feature gate GracefulNodeShutdown"))
 	}
+	if len(kc.PodPriorityShutdownGracePeriods) != 0 && !localFeatureGate.Enabled(features.PodPriorityBasedGracefulShutdown) {
```

Comment:
I think we want one more check, we want to ensure user doesn't pass both `PodPriorityBasedGracefulShutdown` config and existing `ShutdownGracePeriod` config at same time. 

something like

```
if localFeatureGate.Enabled(features.PodPriorityBasedGracefulShutdown) && len(kc.PodPriorityShutdownGracePeriods) != 0 && (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) ==> return an error
```

Reply:
Yes, agree :+1: